src/panels/ActionsPanel/ActionsPanel.tsx

```diff
@@ -85,12 +111,54 @@ export default function ActionsPanel(): JSX.Element {
     appState.juju?.modelData?.[modelUUID as string]?.applications?.[appName]
       ?.charm;
 
-  const generateSelectedUnitList = () => "...";
+  const generateSelectedUnitList = () => "..."; // XXX req unit list selection
 
   const generateTitle = () => (
     <h5>{generateIconImg(appName, namespace)} 0 units selected</h5>
   );
 
+  const executeAction = async () => {
+    // You shouldn't be able to get this far without this defined but jic.
+    if (!selectedAction) return;
+    await executeActionOnUnits(
+      // XXX The unit list is only hard coded until the
+      // unit list selection has been implemented.
+      ["ceph/0"],
+      selectedAction,
+      actionOptionsValues.current[selectedAction],
+      modelUUID,
+      appStore.getState()
+    );
+  };
+
+  const changeHandler = useCallback(
+    (actionName, values) => {
+      onValuesChange(actionName, values, actionOptionsValues);
+      enableSubmit(
+        selectedAction,
+        actionData,
+        actionOptionsValues,
+        setDisableSubmit
+      );
+    },
+    [actionData, selectedAction]
```

Comment:
You've got some missing deps here.

Reply:
I thought so as well but the linter/compiler seem to be happy with this. This was where we were discussing why it doesn't complain about the refs.